File: backend/app/schemas/intervention.py
from datetime import datetime

from pydantic import BaseModel


class InterventionListItem(BaseModel):
    intervention_id: int
    payment_id: int
    action: str
    provider: str | None
    provider_action_id: str | None
    provider_status: str | None
    status: str
    executed_at: datetime | None
    created_at: datetime


class InterventionListResponse(BaseModel):
    items: list[InterventionListItem]
    page: int
    page_size: int
    total: int

File: frontend/src/types/intervention.ts
export interface InterventionLedgerItem {
  intervention_id: number;
  payment_id: number;
  action: string;
  provider: string | null;
  provider_action_id: string | null;
  provider_status: string | null;
  status: string;
  executed_at: string | null;
  created_at: string;
}

export interface InterventionLedgerPage {
  items: InterventionLedgerItem[];
  page: number;
  page_size: number;
  total: number;
}
